### PR TITLE
Role check

### PR DESCRIPTION
**Pull request by NeRpi**

### CODE REVIEW

File: server/src/middleware/jwt.middleware.ts [hunk 1]
```diff
@@ -16,7 +17,7 @@ const jwtMiddleware = (req: Request, res: Response, next: NextFunction) => {
     const userData = tokenService.validateAccessToken(accessToken);
     if (!userData) return next(ApiError.unauthorized("The token is not valid"));
 
-    // req.user = userData;
+    req.user = new UserDto(userData);
```
Comment (NeRpi) [inline server/src/middleware/jwt.middleware.ts]: to do this, I extended the Request type in express/index.d.ts

declare global{
    namespace Express {
        export interface Request {
            user?: UserDto;
        }
    }
}

File: server/src/dto/user.dto.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+export class UserDto {
+  public id?: string;
```
Comment (Daniil-Zaichuk-iTechArt) [inline server/src/dto/user.dto.ts]: all fields are public by default. So you can omit this modifier

File: server/src/middleware/verify.is.super.admin.middleware.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import ApiError from "../error/api.error.ts";
+import { NextFunction, Request, Response } from "express";
+import RoleService from "../services/role.service.js";
+
+const roleService = new RoleService();
+const verifyIsSuperAdminMiddleware = async (
```
Comment (Daniil-Zaichuk-iTechArt) [inline server/src/middleware/verify.is.super.admin.middleware.ts]: Don't think that creating middleware for each role is good idea;)

File: server/src/middleware/verify.is.super.admin.middleware.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  next: NextFunction
+) => {
+  try {
+    const user = req.user;
+    if (!user) return next(ApiError.unauthorized("The user is not logged in"));
+
+    const userRole = await roleService.getById(user.roleId!);
+
+    if (!userRole || userRole.role !== "SuperAdmin")
+      return next(ApiError.internal("Access denied!"));
```
Comment (Daniil-Zaichuk-iTechArt) [inline server/src/middleware/verify.is.super.admin.middleware.ts]: just send 403 Forbidden. No need to throw an error

**Review by Daniil-Zaichuk-iTechArt — APPROVED** [review]

**Review by Daniil-Zaichuk-iTechArt — APPROVED** [review]

File: server/src/repositories/role.repo.ts [hunk 5]
```diff
@@ -1,9 +1,10 @@
 import { RoleEntity } from "../db/entities/role.entity.ts";
 import ApiError from "../error/api.error.ts";
+import { Roles } from "../db/enums/role.enum.js";
 import dbConnector from "../db/db.connector.ts";
 
 export const RoleRepo = dbConnector.getRepository(RoleEntity).extend({
-  async createRole(roleName: string) {
+  async createRole(roleName: Roles) {
```
Comment (Daniil-Zaichuk-iTechArt) [inline server/src/repositories/role.repo.ts]: We want to be able to create new roles not to dublicate existing

File: server/src/services/auth.service.ts [hunk 6]
```diff
@@ -44,7 +44,7 @@ export default class AuthService {
 
   generateToken(userDto: UserEntity) {
     const payload = {
-      userId: userDto.id,
+      id: userDto.id,
```
Comment (Daniil-Zaichuk-iTechArt) [inline server/src/services/auth.service.ts]: I'd name it as userId just to make it clear to which entity it belongs

File: tasks/228.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+//      "a" if a == b
+
+/**
+ * @param {number[]} nums
+ * @return {string[]}
+ */
+var summaryRanges = function (nums) {
+    let result = [];
+    let f = nums.at(0);
+    for (let i = 0; i < nums.length; i++) {
```
Comment (Daniil-Zaichuk-iTechArt) [inline tasks/228.js:23]: Try not to use C-like fors in JS. We have a lot of iteration methods (forEach, reduce, map, etc)

File: tasks/228.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+//      "a->b" if a != b
+//      "a" if a == b
+
+/**
+ * @param {number[]} nums
+ * @return {string[]}
+ */
+var summaryRanges = function (nums) {
+    let result = [];
+    let f = nums.at(0);
```
Comment (Daniil-Zaichuk-iTechArt) [inline tasks/228.js:22]: let f = nums[0]
keep it simple:)

Comment (Daniil-Zaichuk-iTechArt) [reply tasks/228.js:22]: "f" is "first"?

File: tasks/2448.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * @param {number[]} nums
+ * @param {number[]} cost
+ * @return {number}
+ */
+var minCost = function (nums, cost) {
+    let pairArray = nums.map((num, index) => [num, cost[index]]);
+    pairArray.sort((a, b) => a[0] - b[0]);
+    let left = 0;
+    let right = pairArray.length - 1;
```
Comment (Daniil-Zaichuk-iTechArt) [inline tasks/2448.js:23]: JS community doen't have strong opinion about ';', but you definitely need to be consistent with using it

**Review by Daniil-Zaichuk-iTechArt — CHANGES_REQUESTED** [review]

**Review by Daniil-Zaichuk-iTechArt — APPROVED** [review]

**Pull request merged**